sparse/client.go

```diff
@@ -371,25 +390,34 @@ func (client *syncClient) syncHoleInterval(holeInterval Interval) error {
 	return nil
 }
 
-func (client *syncClient) getServerChecksum(checksumInterval Interval) ([]byte, error) {
+func (client *syncClient) getServerChecksum(batchInterval Interval) ([]byte, error) {
 	queries := make(map[string]string)
-	queries["begin"] = strconv.FormatInt(checksumInterval.Begin, 10)
-	queries["end"] = strconv.FormatInt(checksumInterval.End, 10)
+	queries["begin"] = strconv.FormatInt(batchInterval.Begin, 10)
+	queries["end"] = strconv.FormatInt(batchInterval.End, 10)
 	resp, err := client.sendHTTPRequest("GET", "getChecksum", queries, nil)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get checksum")
 	}
 
 	// drain the buffer and close the body
-	checksum, err := ioutil.ReadAll(resp.Body)
+	body, err := ioutil.ReadAll(resp.Body)
 	_ = resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("failed to get server checksum of interval %+v: status code=%v (%v)",
-			checksumInterval, resp.StatusCode, http.StatusText(resp.StatusCode))
+			batchInterval, resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+
+	if err != nil {
+		return nil, err
 	}
 
-	return checksum, err
+	var checksum string
+	if err := json.Unmarshal(body, &checksum); err != nil {
```

Comment:
Does this mean previous `getServerChecksum` always not working correctly? due to a wrong decode.